packages/pwa-kit-extension-sdk/src/configs/webpack/overrides-resolver-loader.ts

```diff
@@ -161,67 +161,63 @@ const OverrideResolverLoader = function (this: LoaderContext<any>) {
  * @returns {boolean} - A boolean indicating if the source file should be processed by the override loader.
  */
 export const validateOverrideSource = (source: string, options: any = {}) => {
-    const {
-        isMonoRepo = false,
-        target = 'node',
-        overridables = [],
-        projectDir = process.cwd()
-    } = options
+    const {isMonoRepo = false, target = 'node', overridables = []} = options
     const isSetupFile = SETUP_FILE_REGEX.test(source)
     const targetCache = OVERRIDABLE_CACHE[target as keyof typeof OVERRIDABLE_CACHE]
 
-    // Extract package path from source to check if it's an extension
-    const folderPattern = `${path.sep}${
-        isMonoRepo ? MONO_REPO_WORKSPACE_FOLDER : NODE_MODULES_FOLDER
-    }${path.sep}`
-    const packagePath = source.split(folderPattern)[1]?.split(path.sep)[0]
-
-    if (!packagePath) {
+    // Exit early if we have:
+    // 1. Processed this file already.
+    // 2. The file is an extension setup file.
+    if (targetCache.includes(source) || isSetupFile) {
         return false
     }
 
-    // Get full package path
-    const fullPackagePath = path.join(
-        projectDir,
-        isMonoRepo ? MONO_REPO_WORKSPACE_FOLDER : NODE_MODULES_FOLDER,
-        packagePath
-    )
+    let projectPath = ''
+    let isExtensionFile = false
+
+    // Check for extension in monorepo packages or node_modules
+    if (source.includes(`${path.sep}${MONO_REPO_WORKSPACE_FOLDER}${path.sep}`)) {
+        // Extract package path from monorepo structure
+        const [basePath, remainingPath] = source.split(
+            `${path.sep}${MONO_REPO_WORKSPACE_FOLDER}${path.sep}`
+        )
+        projectPath = `${basePath}${path.sep}${MONO_REPO_WORKSPACE_FOLDER}${path.sep}${
+            remainingPath.split(path.sep)[0]
+        }`
+    } else if (source.includes(`${path.sep}${NODE_MODULES_FOLDER}${path.sep}`)) {
+        // Special handling for scoped packages (@namespace/package)
+        const [basePath, packagePath] = source.split(`${path.sep}${NODE_MODULES_FOLDER}${path.sep}`)
+        const packageParts = packagePath.split(path.sep)
+
+        projectPath = packageParts[0].startsWith('@')
+            ? `${basePath}${path.sep}${NODE_MODULES_FOLDER}${path.sep}${packageParts[0]}${path.sep}${packageParts[1]}`
+            : `${basePath}${path.sep}${NODE_MODULES_FOLDER}${path.sep}${packageParts[0]}`
+    }
```

Comment:
Is this supposed to be an if/else if.. do we expect that there is a scenario where neither of these conditions are met, or should we be making an assumption that i's always _not_ a mono repo if we can't tell?

The code is also a little hard to follow. Do you think you can either break things down into reusable functions (e.g. the contents of your if and else) .. or maybe flatten the code, it looks like the first couple lines is very similar in the conditions blocks, so we might be able to assign a variable conditionally and then split using that, then only have a special condition for the namespace in an if?

In short, can you take a stap at cleaning this up a little.

Reply:
plus one what Ben said, It would be nice if the code in this section can be simplified a bit 👍